website/app/modifiers/doc-track-event.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Modifier from 'ember-modifier';
+import { registerDestructor } from '@ember/destroyable';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+function handleTriggerEvent(eventName) {
+  // https://usefathom.com/docs/features/events
+  window.fathom?.trackEvent(eventName);
+}
+
+function cleanup(instance) {
+  instance.element.removeEventListener(
+    instance.triggerEvent,
+    handleTriggerEvent
+  );
+
+  instance.element = null;
+  instance.eventName = null;
+  instance.triggerEvent = null;
+}
+
+export default class DocTrackEvent extends Modifier {
+  @service fastboot;
+
+  element = null;
+  eventName = null;
+  triggerEvent = null;
+
+  modify(element, _positional, named) {
+    if (
+      // Only attempt to do something if we are in the right environment
+      this.fastboot.isFastBoot ||
+      window.fathom == null
+    ) {
+      return;
+    }
+
+    const { triggerEvent = 'click', eventName } = named;
+
+    const hasValidEventName = typeof eventName === 'string';
+
+    assert(
+      `@eventName for "doc-track-event" must be a string; received: ${eventName}`,
+      hasValidEventName
+    );
+
+    this.element = element;
+    this.eventName = eventName;
+    this.triggerEvent = triggerEvent;
+
+    element.addEventListener(this.triggerEvent, handleTriggerEvent(eventName));
```

Comment:
I think you can avoid defining an extra variable, no?
```suggestion
    element.addEventListener(triggerEvent, handleTriggerEvent(eventName));
```

That said, there is a more burning problem here: `handleTriggerEvent(eventName)` is an invocation of a function, not a pointer to it, so this line actually executes the `window.fathom?.trackEvent(eventName)` which sends a call to the analytics service, even without the user interacting with the element (if you add a `console.log` to the `handleTriggerEvent` function you will see it).

So this line should be something like this:

```suggestion
    element.addEventListener(triggerEvent, handleTriggerEvent);
```

Now, in order to have make the `eventName` argument available to the `handleTriggerEvent` is to define the function _inside_  the "modifier" call, see for example how it's done here: https://github.com/hashicorp/design-system/blob/main/packages/components/src/modifiers/hds-clipboard.ts#L160-L179

Let me know if you want to pair on this, ok?